### PR TITLE
Adds Namespaces

### LINKED ISSUE #91: Investigate the use of namespaces to improve API structure
MATLAB has a folder based implementation of namespaces [namespaces](https://www.mathworks.com/help/matlab/matlab_oop/scoping-classes-with-packages.html). Can we use this to avoid names like `standardTF_custXY_paraContrasts` and have `standardTF.custXY.paraContrasts` instead?

### PR DESCRIPTION
**Pull request by StephenNneji**
Closes #91

### CODE REVIEW

**Review by DrPaulSharp — COMMENTED** [review]
Looks good so far, will need to look again given changes to calculation code in #116. As far as I can tell though, there aren't any major conflicting changes.

File: tests/testExceptions.m [hunk 1]
```diff
@@ -4,22 +4,22 @@
 % in the RAT utilities.
 %
 % In this class, we test:
-% duplicateName, indexOutofRange, invalidInput, invalidNumberofInputs,
-% invalidOption, invalidType, invalidValue, nameNotRecognised
+% duplicateName, exceptions.indexOutOfRange, invalidInput, exceptions.invalidNumberOfInputs,
+% invalidOption, exceptions.invalidType, exceptions.invalidValue, exceptions.nameNotRecognised
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testExceptions.m:8]: I will remove the list from the comments. It is just another thing to keep updated

File: targetFunctions/+standardTF/layersCore.m [hunk 2]
```diff
@@ -1,5 +1,5 @@
 function [sldProfile,reflect,Simul,shifted_dat,theseLayers,resamLayers,chiSq,ssubs] = ...
-    standardTFLayersCore(contrastLayers, rough, ...
+    layersCore(contrastLayers, rough, ...
```
Comment (DrPaulSharp) [inline targetFunctions/+standardTF/layersCore.m:2]: Given this routine is called by both standardTF and domainsTF layers calculations, I think it is appropriate to remove this function from the standardTF namespace and include it elsewhere.

Comment (DrPaulSharp) [reply targetFunctions/+standardTF/layersCore.m:2]: Also, perhaps the name "coreLayersCalculation" is more helpfully descriptive.

**Review by DrPaulSharp — APPROVED** [review]
All looks good. Given the reorganisation of the targetFunctions directory, I'd propose some similar changes to the API directory:

API/controlsClass/controlsClass.m -> API/controlsClass.m
API/outputFunction/sendTextOutput.m -> API/sendTextOutput.m
API/projectClass/dyLibWrapper.m -> utilities/dyLibWrapper.m (this routine is tested in testUtilities.m)
API/projectClass/mergeStructs.m -> API/mergeStructs.m
API/projectClass/pythonWrapper.m -> utilities/pythonWrapper.m (this routine is tested in testUtilities.m)

I also think we should reconsider the names of some of the enums and rename the subdirectories in "minimisers" to better match their names in the code.

File: targetFunctions/+standardTF/layersCore.m (hunk 2 above)
Comment (StephenNneji) [reply targetFunctions/+standardTF/layersCore.m:2]: My understanding is that domains, runs a couple of standard layers calculation then averages so the coreLayersCalculation belongs to standardTF even though it is used by domains. So I will rename the file but leave it in the same folder

**Comment by StephenNneji** [comment]
I've reverted my extra changes I will make them in another pull request

**Pull request merged**